### PR TITLE
Parse and expose tenc members related to cbcs + tests

### PR DESCRIPTION
**Pull request by SingingTree**

### CODE REVIEW

**Comment by SingingTree** [comment]
Travis failure above appears to be a failure in the CI env rather than a code issue.

**Comment by kinetiknz** [comment]
> Travis failure above appears to be a failure in the CI env rather than a code issue.

Looks like https://github.com/rust-lang/rust/issues/55571, which should be fixed overnight.

File: mp4parse/tests/public.rs [hunk 1]
```diff
@@ -10,8 +10,19 @@ use std::io::{Cursor, Read};
 use std::fs::File;
 
 static MINI_MP4: &'static str = "tests/minimal.mp4";
-static AUDIO_EME_MP4: &'static str = "tests/bipbop-cenc-audioinit.mp4";
-static VIDEO_EME_MP4: &'static str = "tests/bipbop_480wp_1001kbps-cenc-video-key1-init.mp4";
+static AUDIO_EME_CENC_MP4: &'static str = "tests/bipbop-cenc-audioinit.mp4";
+static VIDEO_EME_CENC_MP4: &'static str = "tests/bipbop_480wp_1001kbps-cenc-video-key1-init.mp4";
+// The cbcs files were created via shaka-packager from Firefox's test suite's bibop.mp4 using:
```
Comment (kinetiknz) [inline mp4parse/tests/public.rs]: nit: typo in bipbop

File: mp4parse/src/lib.rs [hunk 2]
```diff
@@ -2154,16 +2159,46 @@ fn read_schi<T: Read>(src: &mut BMFFBox<T>) -> Result<Option<TrackEncryptionBox>
...
+        }
+    };
+    let default_is_encrypted = src.read_u8()?;
     let default_iv_size = src.read_u8()?;
     let default_kid = read_buf(src, 16)?;
+    // If default_is_encrypted == 1 && default_iv_size == 0 we expect a default_constant_iv
+    let default_constant_iv = match (default_is_encrypted, default_iv_size) {
+        (1, 0) => {
+            let default_constant_iv_size = src.read_u8()?;
+            let mut constant_iv: Vec<u8> = Vec::new();
```
Comment (kinetiknz) [inline mp4parse/src/lib.rs]: Is there a reason not to use `read_buf(src, default_constant_iv_size)` here?

Comment (SingingTree) [reply mp4parse/src/lib.rs]: There is not, an oversight on my behalf.

**Comment by SingingTree** [comment]
Updated. I have sent the spec via other channels given its proprietary nature.

**Comment by kinetiknz** [comment]
Thanks!

**Pull request merged**